gettsim/taxes/abgelt_st.py

```diff
@@ -1,33 +1,55 @@
 import numpy as np
+import pandas as pd
 
 
-def abgelt_st(tax_unit, eink_st_params, eink_st_abzuege_params):
+def abgelt_st_m(gem_veranlagt, kind, abgelt_st_m_tu, tu_id):
     """ Capital Income Tax / Abgeltungsteuer
         since 2009, captial income is taxed with a flatrate of 25%.
     """
-    tax_unit["abgelt_st_m"] = 0
-    if eink_st_params["jahr"] >= 2009:
-        tax_unit.loc[~tax_unit["gem_veranlagt"], "abgelt_st_m"] = eink_st_params[
-            "abgelt_st_satz"
-        ] * np.maximum(
-            tax_unit["brutto_eink_5"]
-            - eink_st_abzuege_params["sparerpauschbetrag"]
-            - eink_st_abzuege_params["sparer_werbungskosten_pauschbetrag"],
-            0,
-        ).round(
-            2
+    out = np.select(
+        [gem_veranlagt & ~kind, ~gem_veranlagt & ~kind, kind],
+        [tu_id.replace(abgelt_st_m_tu) / 2, tu_id.replace(abgelt_st_m_tu), 0],
+    )
+    return pd.Series(data=out, index=tu_id.index, name="abgelt_st_m")
+
+
+def abgelt_st_m_tu(_zu_verst_kapital_eink_tu, abgelt_st_params):
+    """
+
+    Parameters
+    ----------
+    _zu_verst_kapital_eink_tu
+    abgelt_st_params
+
+    Returns
+    -------
+
+    """
+    out = abgelt_st_params["abgelt_st_satz"] * _zu_verst_kapital_eink_tu
+    return out.rename("abgelt_st_m_tu")
+
+
+def _zu_verst_kapital_eink_tu(
+    brutto_eink_5_tu, gemeinsam_veranlagte_tu, eink_st_abzuege_params
+):
+    """
+
+    Parameters
+    ----------
+    brutto_eink_5_tu
+    gemeinsam_veranlagte_tu
+    eink_st_abzuege_params
+
+    Returns
+    -------
+
+    """
+    out = (
```

Comment:
Replace `np.select` with `series.replace` and maybe create a temp variable to make the expression less complex.